src/ngraph/runtime/gpu/cudnn_emitter.cpp

```diff
@@ -158,13 +158,34 @@ cudnnDataType_t runtime::gpu::CUDNNEmitter::get_cudnn_datatype(const element::Ty
     return get_cudnn_datatype(dtype.c_type_string());
 }
 
+size_t
+    runtime::gpu::CUDNNEmitter::get_cudnn_workspace_size(const cudnnHandle_t& handle,
+                                                         const cudnnReduceTensorDescriptor_t& desc,
+                                                         const cudnnTensorDescriptor_t& input_desc,
+                                                         const cudnnTensorDescriptor_t& output_desc,
+                                                         size_t input_buffer_size)
+{
+    size_t workspace_size = 0;
+    CUDNN_SAFE_CALL(
+        cudnnGetReductionWorkspaceSize(handle, desc, input_desc, output_desc, &workspace_size));
+    if (workspace_size < input_buffer_size)
```

Comment:
IS this still necessary?

Reply:
Not sure abt that... @csullivan Can you please point me to the tests that captures this case?

Reply:
I would expect this isn't necessary any more. This is the test that revealed the bug `softmax_axis_3d_double`. That said, softmax is now instrumented with cuda, not cudnn, so this test doesn't cover the issue any longer.

Reply:
In that case I can remove this conditional. Also the wrapper function seems to redundant then.